lib/kafka/consumer.rb

```diff
@@ -280,6 +280,21 @@ def each_batch(min_bytes: 1, max_wait_time: 5, automatically_mark_as_processed:
       end
     end
 
+    # @param min_bytes [Integer] the minimum number of bytes to read before
+    #   returning messages from the server; if `max_wait_time` is reached, this
+    #   is ignored.
+    # @param max_wait_time [Integer, Float] the maximum duration of time to wait before
+    #   returning messages from the server, in seconds.
+    # @param topics_offsets [Hash] the offsets to fetch from. By default it will
+    #   fetch from where the offset manager defaults too.
+    # @return batch [Kafka::FetchedBatch] a message batch fetched from Kafka.
+    def poll(min_bytes: 1, max_wait_time: 5, offsets: {})
```

Comment:
@dasch with `Sync` you refer to `commitSync`? 
On our case we want to be able to seek without propagating that offset until we receive the done message and ready to actually commit offsets. 
The downside of not propagating is that if the consumer goes down it could/will double process some messages. I see how using `Sync` would be preferable on most cases. 
 i think we could make `sync` should be optional. WDYT?